### PR TITLE
HAWQ-1184. Fix risky "-Wshift-negative-value, -Wparentheses-equality,-Wtautological-compare" types of compile warnings under osx.

### PR DESCRIPTION
**Pull request by xunzhang**

### CODE REVIEW

**Comment by xunzhang** [comment]
cc @paul-guo-

File: src/backend/catalog/pg_filesystem.c [hunk 1]
```diff
@@ -384,7 +384,7 @@ FileSystemGetNameByOid(Oid	fsysOid)
 
 char *fsys_func_type_to_name(FileSystemFuncType ftype)
 {
-	if(ftype < 0 || ftype >= FSYS_FUNC_TOTALNUM)
+	if (!(ftype >= FSYS_FUNC_CONNECT && ftype < FSYS_FUNC_TOTALNUM))
```
Comment (paul-guo-) [inline src/backend/catalog/pg_filesystem.c]: It does not seem to need a conversion of !. 
The code below seems to apply either unsigned or signed type, which enum may use.
ftype < FSYS_FUNC_CONNECT || ftype >= FSYS_FUNC_TOTALNUM

**Comment by paul-guo-** [comment]
+1

**Comment by xunzhang** [comment]
Merged into master.

**Pull request closed without merging**